libs/s25main/ai/random.h

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2005 - 2025 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "helpers/random.h"
+#include <random>
+
+namespace AI {
+
+std::minstd_rand& getRandomGenerator();
+
+template<typename T>
+T randomValue(T min = std::numeric_limits<T>::min(), T max = std::numeric_limits<T>::max())
+{
+    return helpers::randomValue(getRandomGenerator(), min, max);
+}
+
+// Return a random bool:
+//   random()           ... will return true|false with 50% chance each
+//   random(15)         ... will return true in 1/15 of the cases
```

Comment:
So the intended semantic is: Return true in 1 out of N cases? Then for the 50% case you'd want `2` not `1`, don't you?